components/renderers/controls/array/array-renderer.ts

```diff
@@ -2,80 +2,105 @@
 
 class ArrayRenderer implements JSONForms.IRenderer {
 
+    private maxSize = 99;
+    priority = 2;
+
     constructor(private pathResolver: JSONForms.IPathResolver, private scope: ng.IScope) {
 
     }
 
-    private defaultGridOptions(services: JSONForms.Services, schema: SchemaElement): uiGrid.IGridOptions {
+    private static isExternalFilterAvailable(dataProvider: JSONForms.IDataProvider,
+                                             options: uiGrid.IGridOptions): boolean {
+        return dataProvider.filter !== undefined &&
+            options.hasOwnProperty('useExternalFiltering');
+    }
+
+    private static isExternalPagingAvailable(dataProvider: JSONForms.IDataProvider,
+                                             options: uiGrid.IGridOptions) {
+        return dataProvider.fetchPage !== undefined &&
+                options.hasOwnProperty('useExternalPagination');
+    }
+
+    private defaultGridOptions(existingOptions: uiGrid.IGridOptions,
+                               services: JSONForms.Services, schema: SchemaElement): uiGrid.IGridOptions {
 
-        var paginationEnabled = false;
-        var defaultGridOptions:uiGrid.IGridOptions = {};
-        defaultGridOptions.enableColumnResizing = true;
-        defaultGridOptions.enableHorizontalScrollbar = 0;
-        defaultGridOptions.enableVerticalScrollbar = 0;
         var dataProvider = services.get<JSONForms.IDataProvider>(JSONForms.ServiceId.DataProvider);
         var validationService = services.get<JSONForms.IValidationService>(JSONForms.ServiceId.Validation);
 
-        if (dataProvider instanceof JSONForms.DefaultInternalDataProvider) {
-            defaultGridOptions.totalItems = dataProvider.getTotalItems();
-        } else {
+        let externalPaginationEnabled = ArrayRenderer.isExternalPagingAvailable(dataProvider, existingOptions);
+        let externalFilteringEnabled  = ArrayRenderer.isExternalFilterAvailable(dataProvider, existingOptions);
```

Comment:
It shouldn't be because `defaultGridOptions` is called with `element.options || {}`